wp-includes/widgets/class-wp-widget-media.php

```diff
@@ -234,6 +235,45 @@ class="media-widget-instance-property"
 	}
 
 	/**
+	 * Deletes the Widget when the attachment file is deleted.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param int $post_id Attachment ID.
+	 */
+	public function delete_attachment_data( $post_id ) {
```

Comment:
Outright the widget instances when the associated attachment is deleted makes me nervous. I think it would be better to instead cause the widget to not render anything, and show a notification in the widget control that the attachment has been removed. For example, I think it should behave like an RSS widget pointing to a 404 feed URL.

By showing a notice that the attachment has been deleted, there would be less user mystery when they find a widget disappeared. Deleting the widget would also cause the widget's title to be dropped, resulting in data loss. I think it would be better if a user discovers they deleted an image used by an image widget, and then picks another image for that existing widget, rather than having to create a whole new widget from scratch.